libs/sdk-ui/src/base/measureTitles/ignoreTitlesForSimpleMeasures.ts

```diff
@@ -0,0 +1,75 @@
+// (C) 2007-2020 GoodData Corporation
+import {
+    IInsightDefinition,
+    IBucket,
+    IAttributeOrMeasure,
+    isMeasure,
+    IMeasure,
+    IMeasureDefinitionType,
+    insightBuckets,
+    bucketItems,
+    isMeasureDefinition,
+    isArithmeticMeasureDefinition,
+    isPoPMeasureDefinition,
+    isPreviousPeriodMeasureDefinition,
+} from "@gooddata/sdk-model";
+
+function isAdhocMeasure(definition: IMeasureDefinitionType): boolean {
+    return (
+        isMeasureDefinition(definition) &&
+        (!!definition.measureDefinition.aggregation ||
+            !!definition.measureDefinition.computeRatio ||
+            (Array.isArray(definition.measureDefinition.filters) &&
+                definition.measureDefinition.filters.length > 0))
+    );
+}
+
+function isNonSimpleMeasure(definition: IMeasureDefinitionType): boolean {
+    return (
+        isArithmeticMeasureDefinition(definition) ||
+        isPoPMeasureDefinition(definition) ||
+        isPreviousPeriodMeasureDefinition(definition)
+    );
+}
+
+function removeTitleForSimpleMeasure(bucket: IBucket): IBucket {
+    const items = bucketItems(bucket);
+    return {
+        ...bucket,
+        items: items.map(
+            (bucketItem: IAttributeOrMeasure): IAttributeOrMeasure => {
+                if (
+                    isMeasure(bucketItem) &&
+                    !isAdhocMeasure((bucketItem as IMeasure).measure.definition as IMeasureDefinitionType) &&
+                    !isNonSimpleMeasure((bucketItem as IMeasure).measure.definition as IMeasureDefinitionType)
+                ) {
+                    delete (bucketItem as IMeasure).measure.title;
+                }
+                return bucketItem;
+            },
+        ),
+    };
+}
+
+/**
+ * This function ignores the titles of simple measures.
+ *
+ * For simple measures, their titles are removed.
+ * For adhoc or non-simple measures, their titles are left intact.
+ *
+ * @param {IInsight|IInsightDefinition} insight - insight or insight definition that must be processed.
+ *
+ * @returns {IInsight}
+ *
+ * @internal
+ */
+export function ignoreTitlesForSimpleMeasures<T extends IInsightDefinition>(insight: T): T {
+    const buckets = insightBuckets(insight);
+    // tslint:disable-next-line:no-object-literal-type-assertion
+    return {
+        insight: {
+            ...insight.insight,
+            buckets: buckets.map((bucket: IBucket): IBucket => removeTitleForSimpleMeasure(bucket)),
+        },
+    } as T;
+}
```

Comment:
this needs to be generalized and most of the code moved to sdk-model. a model method that aids mass-modification of insight buckets is definitely helpful. instead of this to-the-point approach, we should have something like:

insightModifyItems(insight, modifyFunction).

the method will take care of:
1. iteration over buckets
2. iteration over items
3. calling the modifyFunction

the contract of modifyFunction is, that whatever it returns will be used instead of the original item.

4. create new insight definition with modified buckets

---

ideally, steps 1+2 above would be further handled by bucketsModifyItem(buckets, modifyFunction) and bucketModifyItems(bucket, modifyFunction) so that the modification functionality can be used from multiple places.

---

additionally, please change the title removal code so that it uses the existing modifyMeasure function `modifyMeasure(measure, m => m.noTitle());` << this will drop the title

Reply:
Thanks, I've updated the PR.